pkg/controller/networkpolicy/validate.go

```diff
@@ -366,6 +366,13 @@ func (t *tierValidator) updateValidate(curObj, oldObj interface{}, userInfo auth
 	reason := ""
 	curTier := curObj.(*secv1alpha1.Tier)
 	oldTier := oldObj.(*secv1alpha1.Tier)
+	// Allow an exception of Emergency Tier Priority update from 5 to 20 as we downgrade its priority intentionally
+	// from antrea-controller.
+	if curTier.Name == emergencyTierName {
+		if curTier.Spec.Priority == 20 && oldTier.Spec.Priority == 5 {
+			return "", true
+		}
+	}
```

Comment:
Is there a more elegant way of doing this by checking if `userInfo` matches the controller's service account?

Reply:
if we rely on the user info .. then anyone assigned the ClusterRole can update the Tier priority to any value.. the cluster admin controls this.. so it should be fine.  but just want to confirm if it is okay to do so?

Reply:
Is it really related to the ClusterRole?

I was thinking of something like this:

1) use downward API to define an env variable with serviceAccountName for the controller Pod:

```
        - name: ANTREA_CONTROLLER_SERVICE_ACCOUNT
          valueFrom:
            fieldRef:
              fieldPath: spec.serviceAccountName
```

2) retrieve service account name in validating webhook

```go
serviceAccountName := env.get("ANTREA_CONTROLLER_SERVICE_ACCOUNT")
if serviceAccountName == "" {
    serviceAccountName = "antrea-controller" // default value, supports antrea-controller as a process
}
```

3) make sure userInfo matches serviceAccountName

```go
if serviceaccount.MatchesUsername("kube-system", serviceAccountName, userInfo.Username) { // "kube-system" should not be hardcoded
    // GOOD
} else {
    // BAD
}
```

The fact that [MatchesUsername](https://github.com/kubernetes/apiserver/blob/eb2fbd33936886a70868e97b130afd1e5cd40389/pkg/authentication/serviceaccount/util.go#L55) exists makes me believe that it's not a totally unusual pattern. This seems specific to the serviceAccount and independent of the ClusterRole?

Reply:
you are correct.. you said serviceAccount, my mind registered ClusterRole..